Feat/new doc

hcfw007 opened the pull request.

su-chang commented:
@hcfw007 please resolve the conflicts, thanks.

hcfw007 commented:
> @hcfw007 please resolve the conflicts, thanks.

The conflicts will be resolved once the i18n pr is merged...

huan commented:
Thanks for refactoring the docs with the API!

I'm still thinking about we should keep the original directory structure (`/api/` instead of the `wechaty-api`) and add the `create-puppet` to another folder because the `/api/` is for the Wechaty SDK users, and the `create-puppet` is for the Wechaty SDK developers.

Is there any reason strong enough to rename the `api` to `wechaty-api`? If not, let's thinking about how to use the original `api` folder, and find another better place for the `create-puppet` folder.

Thanks!

hcfw007 commented:
I changed the directory structure from ```api``` to ```wechaty-api``` because I thought there could be other documents to be appended, e.g. ```grpc-api```, ```puppet-api```, for those who wish to contribute to other projects.

As we moved ```puppet-api``` to ```create-puppet``` now, maybe we can change the name back to ```api```, and find better names for people wish to contribute to other wecahty projects?

huan commented:
Thanks for the explanation!

I understand the reason you change the name.

Could you please rename it back to `API`, and let's create another folder named `Puppet API` to contain other docs?

For this structure, the `API` will contain all the Wechaty API for end users, and the `Puppet API` will be the related API for Puppet Developers.

hcfw007 commented:
@huan I have change the dir structure according to your suggestion, and translated one of the pages. Please check it out.

huan commented on docusaurus/docs/api/contact-self.md:
Why do we remove this doc?

hcfw007 replied:
done

huan replied:
May I know what's the reason removed it before?

huan commented:
@hcfw007 Please get two approvals from our contributors before I dive into this big PR; thanks!

hcfw007 commented:
> @hcfw007 Please get two approvals from our contributors before I dive into this big PR; thanks!

@su-chang @iamrajiv @windmemory

windmemory reviewed (dismissed):
LGTM

huan commented:
Can we resolve the conflicts first before approve a PR? Thanks!

hcfw007 commented:
> Can we resolve the conflicts first before approve a PR? Thanks!

The conflicts are resolved.

I'd like to speak to defend these riviews. This PR is marked ready and I ping them to review on Jun 2, 2023, 7:36 PM GMT+8, and the room-invitation PR was accpeted and merged on Jun 3, 2023, 12:19 AM GMT+8.

windmemory reviewed (dismissed):
LGTM

huan commented on docusaurus/docs/api/contact-self.md:
```diff
@@ -1,121 +1,100 @@
 ---
-title: ContactSelf
+title: Contact-Self
+sidebar_label: ' Contact-Self'
```
Use class name as well.

hcfw007 replied:
ditto

hcfw007 replied:
done

huan commented on docusaurus/docs/api/contact-self.md:
````diff
@@ -1,121 +1,100 @@
+## Contact-Self Class
 
-## ContactSelf
+Contact-Self is a special kind of contact. It represents bot self. It has all methods and properties of a regular contact, and some more. Every method that returns a ```ContactInterface```, if the contact is bot self, a ```ContactSelfInterface``` will be returned instead.
````
We should keep using the class name. 

May I know why you change to the folder name `Contact-Self`?

hcfw007 replied:
ditto

hcfw007 replied:
done

huan commented on docusaurus/docs/api/contact-self.md:
````diff
@@ -1,121 +1,100 @@
-
-GET / SET bot avatar
+```ts
+const contact = await bot.Contact.find({id: 'contactId-0' }) // contact0
````
The `find` is inherited from the `Contact` class.

Why do we introduce this code example in the `ContactSelf` extended class?

hcfw007 replied:
The ```find``` in ```ContactSelf``` is not just inhereited from ```Contact``` class. 

See https://github.com/wechaty/wechaty/blob/8bec2d5a66994d7c702c9f9c4fba3f5792273c87/src/user-modules/contact-self.ts#L57

huan replied:
Yes, I remember this because I think I wrote it.

However, the propose of its existence for it is to prevent the abuse of the `ContactSelf.find()` because it is meanling less: there will always be one `ContactSelf` instance for one Wechaty instance.

So it will be also meaning less to introduce this API in the docs.

The best practice of getting a `ContactSelf` instance is to monitor the `login` event and get it from the call back argument.

hcfw007 replied:
It is indeed. So can we keep this introduction on static method ```find```, because it's there, and explain why it should not be used and we should get ```ContactSelf``` instance from ```login``` event listener?

hcfw007 replied:
I have appended with the explaination above, please check it out.

huan commented on docusaurus/docs/api/contact-self.md:
````diff
@@ -1,121 +1,100 @@
-| :--- | :--- |
-| \[file\] | `FileBox` |
+```ts
+static async find (query : string | PUPPET.filters.Contact): Promise<undefined | ContactInterface> 
````
`find` here is a method from the parent class

hcfw007 replied:
ditto

huan commented on docusaurus/docs/api/contact-self.md:
```diff
@@ -1,121 +1,100 @@
-// Save avatar to local file like `1-name.jpg`
+Example:
 
-bot.on('login', async user => {
```
The code here is the proper documentation.

What's the reason we deleted them?

hcfw007 replied:
Becuase in the new doc, this section is the code example for the static method find.

The ContactSelf part is complete rewroten, so I recommend to read the new doc directly, instead of reading the diff.

hcfw007 commented on docusaurus/docs/api/contact-self.md:
```diff
@@ -1,121 +1,100 @@
 ---
-title: ContactSelf
+title: Contact-Self
```
I'll change that, and also for roomInvitation. 

Instead of editing on the old doc, this doc is a fresh start fro scratch. So I did not changed it on puporse.

hcfw007 replied:
In the old doc, ContactSelf was refrenced as one word ```ContactSelf```, and RoomInvitation was refrenced as two words ```Room Invitation```. Could you tell which one is prefered?

hcfw007 replied:
done

huan commented on docusaurus/docs/references/overview.mdx:
```diff
@@ -4,14 +4,11 @@ title: Wechaty References Overview
 ## References Index
 
-1. [API References](../api/overview.mdx)
+1. [API References](../references/overview.mdx)
```
Why did we change the link here?

huan commented on docusaurus/docs/references/overview.mdx:
```diff
@@ -4,14 +4,11 @@ title: Wechaty References Overview
 sidebar_label: 'Overview'
 ---
 
-import { Toc as ApiToc }  from '../api/overview.mdx'
```
Why did we remote those Toc here?

huan commented on docusaurus/docs/wechaty.mdx:
```diff
@@ -26,7 +26,7 @@ Wechaty offers out-of-the-box support for the common features you would need for
 :::note API References
 
-Learn more functions from [API Reference](./api/overview.mdx) and [How-to Guides](./howto/overview.mdx).
+Learn more functions from [API Reference](./references/overview.mdx) and [How-to Guides](./howto/overview.mdx).
```
ditto

huan commented on docusaurus/docs/api/overview.mdx:
Why do we remove this file?

huan commented on docusaurus/docs/api/wechaty.md:
Why did we remove this file?

huan reviewed (changes requested):
Can we split this huge PR into four parts?

1. The modified `docs/api/`
2. The added `docs/puppet-api`
3. the added `docs/i18n/`
4. other misc modifications

After that, I think I will be able to review it and we can fix them separately instead of this whole mess.

Thank you!

hcfw007 commented:
> Can we split this huge PR into four parts?
> 
> 1. The modified `docs/api/`
> 2. The added `docs/puppet-api`
> 3. the added `docs/i18n/`
> 4. other misc modifications
> 
> After that, I think I will be able to review it and we can fix them separately instead of this whole mess.
> 
> Thank you!

OK
